crates/sui-bridge-indexer/src/eth_worker.rs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::postgres_manager::{write, PgPool};
+use crate::{BridgeDataSource, TokenTransfer, TokenTransferData, TokenTransferStatus};
+use ethers::providers::Provider;
+use ethers::providers::{Http, Middleware};
+use ethers::types::Address as EthAddress;
+use std::sync::Arc;
+use sui_bridge::abi::{EthBridgeEvent, EthSuiBridgeEvents};
+use sui_bridge::types::EthLog;
+
+pub async fn process_finalized_eth_events(
+    mut eth_events_rx: mysten_metrics::metered_channel::Receiver<(EthAddress, u64, Vec<EthLog>)>,
+    provider: Arc<Provider<Http>>,
+    pool: &PgPool,
+) {
+    while let Some(event) = eth_events_rx.recv().await {
+        for log in event.2.iter() {
```

Comment:
instead of `event.2.iter()`, you can destructure `event`

```suggestion
    while let Some((_, _, logs)) = eth_events_rx.recv().await {
        for log in logs.iter() {
```